### ISSUE TITLE
twisted.words setup metadata is not up to date anymore

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6402]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#6402|
|Type|defect|
|Created|2013-04-02 23:12:37Z|
|Branch|https://github.com/twisted/twisted/tree/words-setup-metadata-6402|

[twisted/words/topfiles/setup.py](twisted/words/topfiles/setup.py)'s `long_description` and `classifiers` talk about MSN, OSCAR (AIM & ICQ), TOC (AOL), and X-Ish. All gone by now.

Attachments:

* [twisted-words-topfiles-setup.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/74a/74a1896931c3df223fadbb6f7bc4972c534a0b34/736dd10ee3718b839dffa385e9d8b56d16b58710.patch) (598 bytes) - added by nothung on 2013-04-17 01:01:12Z - patch to Ticket #6402, updates long_description in twisted/words/topfiles/setup.py
* [twisted-words-topfiles-setup-v2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/74a/74a1896931c3df223fadbb6f7bc4972c534a0b34/ccd9d361275156218122c13e0e385a962e32432a.patch) (1002 bytes) - added by nothung on 2013-04-18 01:46:06Z - (revised) patch to Ticket #6402, updates long_description in twisted/words/topfiles/setup.py , (new to this revision) text is reflowed to meet Twisted coding standards

<details><summary>Searchable metadata</summary>

```
trac-id__6402 6402
type__defect defect
reporter__thijs thijs
priority__normal normal
milestone__ 
branch__branches_words_setup_metadata_6402 branches/words-setup-metadata-6402
branch_author__rwall rwall
status__closed closed
resolution__fixed fixed
component__words words
keywords__documentation_easy_gsoc documentation easy gsoc
time__1364944357000000 1364944357000000
changetime__1366492244000000 1366492244000000
version__None None
owner__nothung nothung
cc__thijs cc__ralphm
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_2"></a>|@ralphm commented|
|-|-|

OSCAR support is still there though, and the XMPP support in Words is built on top of xish.

**Comment by twisted-trac** [comment]
|<img alt="nothung's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|nothung set owner to nothung<br>nothung set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="nothung's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|nothung commented|
|-|-|

The above patch removes mentions of "MSN" and "TOC (AOL)" from long_description.  Kept mentions of "OSCAR (AIM & ICQ)" and "X-ish" following ralphm's suggestion.

Please review this patch for the Outreach Program for Woman (OPW).  Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess removed owner<br>@jesstess set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_6"></a>|@tomprince set owner to nothung|
|-|-|

This looks good, but the description probably wants to be reflowed.

**Comment by twisted-trac** [comment]
|<img alt="nothung's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|nothung removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_9"></a>|@wallrj set owner to @wallrj<br>@wallrj set status to `assigned`|
|-|-|

reviewing...

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_10"></a>|@wallrj commented|
|-|-|

(In [38194]) Branching to 'words-setup-metadata-6402'

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_11"></a>|@wallrj commented|
|-|-|

(In [38195]) apply twisted-words-topfiles-setup-v2.patch refs [#6402](https://github.com/twisted/twisted/issues/6402)

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_12"></a>|@wallrj set owner to nothung<br>@wallrj set status to `new`|
|-|-|

Code Review:

Thanks nothung for rewrapping and submitting a second patch.

I've added it to a branch and fixed a couple of things:
 * log:branches/words-setup-metadata-6402

 1. There was some trailing whitespace in the patch which I have
    removed. Try setting up your editor to remove trailing whitespace.

    See https://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html#auto5

 2. I added an empty .misc news file - always include a news file in
    your patches. Although empty files can be difficult with svn diff.

    See https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles

Otherwise this looks fine and I'll merge it to trunk.

-RichardW.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_13"></a>|@wallrj set status to `closed`|
|-|-|

(In [38198]) 
Merge words-setup-metadata-6402

Author: nothung
Reviewer: tomprince,rwall
Fixes: [#6402](https://github.com/twisted/twisted/issues/6402)

Remove mention of obsolete protocols from the twisted.words setup.py
metadata.
